src/providers/subgraph-provider.ts

```diff
@@ -22,7 +20,41 @@ export interface ISubgraphProvider<TSubgraphPool extends SubgraphPool> {
 
 const PAGE_SIZE = 1000; // 1k is max possible query size from subgraph.
 
-export abstract class SubgraphProvider {
+export type V3V4SubgraphPool = {
+  id: string;
+  feeTier: string;
+  liquidity: string;
+  token0: {
+    id: string;
+  };
+  token1: {
+    id: string;
+  };
+  tvlETH: number;
+  tvlUSD: number;
+};
+
+export type V3V4RawSubgraphPool = {
+  id: string;
+  feeTier: string;
+  liquidity: string;
+  token0: {
+    symbol: string;
+    id: string;
+  };
+  token1: {
+    symbol: string;
+    id: string;
+  };
+  totalValueLockedUSD: string;
+  totalValueLockedETH: string;
+  totalValueLockedUSDUntracked: string;
+};
+
+export abstract class SubgraphProvider<
+  TRawSubgraphPool extends V3V4RawSubgraphPool,
```

Comment:
is this the right way to go about it? because technically subgraph pool can also be from v2 right?

Reply:
i understand its a replcament but curious why we did not include v2 in this subgraphProvider originally

Reply:
Because I think it's not worth the effort to refactor v2 subgraph. V3 and V4 subgraph pools have similar data structures like ticks, fee tiers, etc., meanwhile V2 goes with reserves. In future, we are more likely to stay with ticks, fee tiers, etc., for concentrated liquidities, which leaves V2 a special case.